src/tools/pub/cmd/pub-add.md

```diff
@@ -39,6 +39,26 @@ $ dart pub add foo:'>2.0.0 <3.0.1'
 If `dart pub add <package>:<constraint>` is an existing dependency,
 it will update the constraint.
 
+## Adding Dependency Overrides
+
+Starting in Dart 3, `dart pub add` supports adding dependency overrides in a similar fashion to dev dependencies. To add a dependency override, use the following syntax:
+`dart pub add override:<package>:<version>`
```

Comment:
This is pretty good. I think we can drop the first sentence though, since I don't think users are familiar with the new way to add dev dependency yet anyway. 

Then we want to link to the dependency override documentation, so they can learn the specific details and potential concerns with them. That way we don't have to repeat too much here. The dev dependency section does the same thing.

Also we need to be careful as users don't necessarily need to specify a version, they could specify any of the [source descriptors](https://dart.dev/tools/pub/cmd/pub-add#source-descriptor). How about something like this:

```suggestion
To specify a [dependency override][], add the `override:` prefix and
specify any [source descriptor](#source-descriptor).

[dependency override]: /tools/pub/dependencies#dependency-overrides
```